pkg/connector/connector.go

```diff
@@ -0,0 +1,86 @@
+// mautrix-twitter - A Matrix-Twitter puppeting bridge.
+// Copyright (C) 2024 Tulir Asokan
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+package connector
+
+import (
+	"context"
+	"fmt"
+
+	"maunium.net/go/mautrix/bridgev2"
+	"maunium.net/go/mautrix/bridgev2/database"
+)
+
+type TwitterConnector struct {
+	br *bridgev2.Bridge
+
+	Config *Config
+}
+
+var _ bridgev2.NetworkConnector = (*TwitterConnector)(nil)
+
+func NewConnector() *TwitterConnector {
+	return &TwitterConnector{}
+}
+
```

Comment:
The `NewConnector` function is an older pattern that isn't used anymore (the cmd/.../main.go should just initialize the struct directly)

```suggestion
```